cu29/src/curuntime.rs

```diff
@@ -217,9 +240,12 @@ pub fn compute_runtime_plan(config: &CuConfig) -> CuResult<Vec<CuExecutionStep>>
             output_msg_type,
             culist_output_index,
         };
-        result.push(step);
+        result.push(CuExecutionUnit::Step(step));
     }
-    Ok(result)
+    Ok(CuExecutionLoop {
+        steps: result,
+        loop_count: None, // if in not unit testing, the main loop is infinite
```

Comment:
It'd be helpful to have a of node that can cause the execution to early-exit. 

This is useful for tests, simulations, etc that are meant to exit on a given condition in either a clean exit (return code 0) or an error exit (return code non-zero).

This would allow you to remove `loop_count` entirely and only allow the execution to stop on a specific condition or with Ctrl+C

Reply:
but here we are at compile time, this is just to represent the overall structure to execute, not really to execute it itself.

But I get what you mean, we could add a "debug" feature that can at runtime override the exit of the plan to do a kind of conditional breakpoint of some sort. Let's add that to the todo list.